gusb/gusb-device.c

```diff
@@ -916,29 +916,32 @@ g_usb_device_get_string_descriptor (GUsbDevice  *device,
 }
 
 /**
- * g_usb_device_get_string_descriptor_bytes:
+ * g_usb_device_get_string_descriptor_bytes_full:
  * @desc_index: the index for the string descriptor to retrieve
  * @langid: the language ID
+ * @length: size of the request data buffer
  * @error: a #GError, or %NULL
  *
  * Get a raw string descriptor from the device. The returned string should be freed
  * with g_bytes_unref() when no longer needed.
  *
  * Return value: (transfer full): a possibly UTF-16 string, or NULL on error.
  *
- * Since: 0.3.6
+ * Since: 0.3.8
  **/
 GBytes *
-g_usb_device_get_string_descriptor_bytes (GUsbDevice  *device,
-					  guint8       desc_index,
-					  guint16      langid,
-					  GError     **error)
+g_usb_device_get_string_descriptor_bytes_full (GUsbDevice  *device,
+					       guint8       desc_index,
+					       guint16      langid,
+					       gsize        length,
+					       GError     **error)
 {
 	gint rc;
 	unsigned char buf[128];
 
 	g_return_val_if_fail (G_USB_IS_DEVICE (device), NULL);
 	g_return_val_if_fail (error == NULL || *error == NULL, NULL);
+	g_return_val_if_fail (length <= 128, NULL);
```

Comment:
That seems like an arbitrary restriction. I think it's better just to allow any `length` and g_autofree g_malloc0 the  `buf`. If a device want a 256 byte descriptor that should be fine.